### PR TITLE
Clicking a label doesn't always focus the form control

### LINKED ISSUE #1514: [form-field] Clicking a label does not always focus the form control
### Provide a general summary of the issue here

For text field and text area, it properly focuses the input when you click on the label. However for select and radiogroup this doesn't seem to work.

### 🤔 Expected Behavior?

-

### 😯 Current Behavior

-

### 💁 Possible Solution

_No response_

### 🖥️ Steps to Reproduce

-

### What browsers are you seeing the problem on? (only for bugs in code)

Chrome

### If other, please specify.

_No response_

### What operating system are you using?

macOS

### 👤 Your name

Jeroen Zwartepoorte

### 🧢 Your product/team

SLDS

### 🎨 Your Theme(s)

Sanoma Learning

### PR DESCRIPTION
**Pull request by jpzwarte**
Fixes #1514

### CODE REVIEW

File: packages/components/radio-group/src/radio-group.ts [hunk 1]
```diff
@@ -185,6 +186,12 @@ export class RadioGroup<T = unknown> extends FormControlMixin(LitElement) {
     this.#rovingTabindexController.focus();
   }
 
+  #onClick(event: Event): void {
+    if (event.target === this) {
```
Comment (anna-lach) [inline packages/components/radio-group/src/radio-group.ts:190]: sometimes radio group is getting focus, sometimes not when clicking on the label
https://kind-plant-088739303-1536.westeurope.1.azurestaticapps.net/?path=/story/form-form--all

Comment (anna-lach) [reply packages/components/radio-group/src/radio-group.ts:190]: same with `select`

Comment (jpzwarte) [reply packages/components/radio-group/src/radio-group.ts:190]: Discussed on slack: this is the browser determining if `:focus-visible` applies.

**Review by anna-lach — APPROVED** [review]
Let's wait for @Diaan's review as well :)

**Review by Diaan — APPROVED** [review]

**Pull request merged**